openbsd/OpenBSDMachine.c

```diff
@@ -139,18 +140,7 @@ static void OpenBSDMachine_scanMemoryInfo(OpenBSDMachine* this) {
    }
 
    super->totalMem = uvmexp.npages * this->pageSizeKB;
-   super->usedMem = (uvmexp.npages - uvmexp.free - uvmexp.paging) * this->pageSizeKB;
-
-   // Taken from OpenBSD systat/iostat.c, top/machine.c and uvm_sysctl(9)
-   const int bcache_mib[] = { CTL_VFS, VFS_GENERIC, VFS_BCACHESTAT };
-   struct bcachestats bcstats;
-   size_t size_bcstats = sizeof(bcstats);
-
-   if (sysctl(bcache_mib, 3, &bcstats, &size_bcstats, NULL, 0) < 0) {
-      CRT_fatalError("cannot get vfs.bcachestat");
-   }
-
-   super->cachedMem = bcstats.numbufpages * this->pageSizeKB;
```

Comment:
Why is this part of the code removed?

Reply:
I'm not sure what purpose the cached memory served, as I wasn't seeing that printed to the terminal as it's own stat in specific, although it does effect the amount of used ram reported, which is part of what made it not match up with top(1).

Reply:
There'S a later place in the code that calculates the used memory shown in the Memory meter. Also the cached memory (i.e. file system cache) is not directly printed there, but shown as blue (file buffers/shared memory) or brown (volatile file system cache).

htop tries to track some more resources separately, thus things might seem unintuitive at first.

Reply:
So do you consider the used/free memory not matching top(1) to be intentional? If not, we will need to fix that, otherwise, this code can be put back and not much else needs changing at this point.

Reply:
The intended display is used+compressed+buffers+shared for the used value in the memory bar; the cached value is ignored intentionally, as that's memory that can usually be reclaimed by the kernel if needed. And although cached resides in RAM, it's not usually considered as "used" for htop due to this volatile nature.

used: Memory used by programs, libraries, data
compressed: Memory compressed by te kernel to avoid swapping (e.g. zram)
buffers: File buffers & dirty pages for I/O
shared: Shared memory segments, memory used e.g. by tmpfs
cached: volatile file system cache (e.g. recently read data to optimize disk access)

Example from Linux:
![image](https://github.com/htop-dev/htop/assets/111196/79a1cd74-b4f1-49eb-9abc-ad402b3c7a16)
Although there's practically no memory unused by Linux, the meter declares the more reasonable value of ~20GiB of memory that cannot easily be dropped, but would need to be placed somewhere else if some new program wanted to use too much memory.

Reply:
I do want to make this a different pull request so this doesn't get conflated, but I think it would be good if NetBSD used code which was more in line with what OpenBSD is doing with the sysctl(), VM_UVMEXP, and swapctl(). NetBSD could be using near identical code for these things to reduce needed maintenance, but instead it is using sysctl() and the VM_UVMEXP2 struct, which is like the VM_UVMEXP struct but also covers what OpenBSD is doing to get swap related info with swapctl(). We'll cross that bridge when we get there, perhaps I'll link to this comment from my new pull request once this pull request is done.

Reply:
I almost forgot about this part; did you want me to add back the code logic I took out here? Again, this will make it not print the correct amount of ram and it will no longer align with what OpenBSD's stock `top` cli prints for ram.

Reply:
Given that this change drops initializing the `super->cachedMem` member of the `Machine` structure, I'm inclined to have this block of code stay or be updated and follow the semantics used within `htop`.

The memory metrics in `htop` mostly fall into 4 categories (simplified):
1. used: RAM used by a program for working memory, code or other resources
2. shared: RAM shared between processes, like `shmem` or `mmap`'d files
3. buffers: RAM allocated for write-back buffers of the filesystem
4. caches: RAM used as filesystem (read) cache that could be dropped at any moment

Of those categories 1 and 2 are subject to potential swapping, while 4 is (mostly) never swapped. Category 3 is basically what you'd first write to disk, if you were to need some more free RAM.

This is much simplified, and differs slightly from what `top` displays regarding the handling of shared memory to accommodate for things like `tmpfs`/`ramfs` on Linux.

AFAICS, the code that was removed should have worked in this regard and if there's a need for that code to change I'd recommend to split out that change as a separate PR to separate build fixes from behavioral changes.